In [6]:
import pandas as pd
import numpy as np 
import pyreadstat
pd.set_option('display.max_columns', 1000) #replace n with the number of columns you want to see completely
pd.set_option('display.max_rows', 1000) #replace n with the number of rows you want to see completely

In [7]:
cbc_dataset_sav, metadata = pyreadstat.read_sav('.\CBC_Dataet_main\Dataset (NVP-Index).sav')

In [8]:
cbc_dataset_meandeley = pd.read_csv(".\CBC_Dataet_main\CBC data_for_meandeley_preded.csv")



In [9]:
cbc_dataset_infomation = pd.read_excel(".\CBC_Dataet_main\cbc information.xlsx")
cbc_dataset_pregnant = pd.read_excel(".\CBC_Dataet_main\Dataset for Reference Intervals of Complete Blood Count and Coagulation Tests in Vietnamese Pregnant Women.xlsx")
cbc_dataset_dengue = pd.read_excel(".\CBC_Dataet_main\dengue patients_data set.xlsx")
cbc_dataset_kenya = pd.read_excel(".\CBC_Dataet_main\good_RRP data Kenya haematology_data repository.xlsx")



In [10]:
cbc_dataset_diagnoses = pd.read_csv(".\CBC_Dataet_main\cbc_diagnoses_dataset_with_labels.csv")

---
<h1> disease prepare Data

In [11]:
cbc_dataset_dengue

id  age     sex date of admission date of first symptom  \
0       1   21    male        2017-05-16            2017-05-11   
1       1   21    male        2017-05-16            2017-05-11   
2       1   21    male        2017-05-16            2017-05-11   
3       1   21    male        2017-05-16            2017-05-11   
4       1   21    male        2017-05-16            2017-05-11   
...   ...  ...     ...               ...                   ...   
1977  739   19    male        2018-09-24            2018-09-22   
1978  740   22    male        2018-09-24            2018-09-21   
1979  740   22    male        2018-09-24            2018-09-21   
1980  741   18    male        2018-09-26                   NaT   
1981  742   18  female        2018-09-26            2018-09-21   

            clinical_diag diagnosis  fatal_outcome  tbil    ast    alt    bun  \
0                  Dengue    dengue              0   NaN    NaN  122.0  10.67   
1                  Dengue    dengue              0   NaN    NaN  122.0  10.67   
2                  Dengue    dengue              0   NaN    NaN  122.0  10.67   
3                  Dengue    dengue              0   NaN    NaN  122.0  10.67   
4                  Dengue    dengue              0   NaN    NaN  122.0  10.67   
...                   ...       ...            ...   ...    ...    ...    ...   
1977  Others (infectious)      none              0   NaN  141.0   37.0  13.45   
1978  Others (infectious)      none              0   NaN  113.0   54.0   9.53   
1979  Others (infectious)      none              0   NaN  113.0   54.0   9.53   
1980  Others (infectious)      none              0   NaN    NaN    NaN    NaN   
1981  Others (infectious)      none              0   NaN    NaN    NaN    NaN   

      creatinin  crp  procalcitonin date of sampling sampling Time   wbc  \
0          0.63  0.6            1.7       2017-05-16      16:37:18  2.98   
1          0.63  0.6            1.7       2017-05-18      10:26:53  3.04   
2          0.63  0.6            1.7       2017-05-18      10:28:08  4.84   
3          0.63  0.6            1.7       2017-05-19      11:09:47  6.39   
4          0.63  0.6            1.7       2017-05-19      11:11:02  7.29   
...         ...  ...            ...              ...           ...   ...   
1977       0.90  0.6            0.1       2018-09-26      15:07:09  7.07   
1978       0.80  0.5            0.1       2018-09-25      11:29:46  1.79   
1979       0.80  0.5            0.1       2018-09-26      15:08:25  2.02   
1980        NaN  0.1            0.1       2018-09-26      14:06:16  9.01   
1981        NaN  0.6            0.1       2018-09-26      14:07:31  7.51   

      neut%  lymp%  mono%  eosino%  neut  lymp  mono  eosion   rbc   hb  plt  \
0      10.8   77.9    6.0      4.0  0.32  2.32  0.18    0.12  4.53  131   19   
1      20.0   63.2   11.2      3.3  0.61  1.92  0.34    0.10  4.74  136   20   
2      18.2   51.0   16.9     11.2  0.88  2.47  0.82    0.54  4.44  128   18   
3      20.6   45.2   12.8     19.7  1.31  2.89  0.82    1.26  4.53  130   25   
4      25.5   39.8    9.7     24.3  1.86  2.90  0.71    1.77  4.23  122   33   
...     ...    ...    ...      ...   ...   ...   ...     ...   ...  ...  ...   
1977   50.1   34.2   15.0      0.1  3.54  2.42  1.06    0.01  5.67  164   19   
1978   62.5   30.2    6.7      0.6  1.12  0.54  0.12    0.01  5.07  149   52   
1979   52.4   32.7   13.9      0.0  1.06  0.66  0.28    0.00  4.87  141   40   
1980   37.2   38.3   12.1     12.1  3.35  3.45  1.09    1.09  5.00  140   14   
1981   60.4   28.0    9.9      1.6  4.54  2.10  0.74    0.12  5.00  140  289   

      ipf%  ipf#   hct  \
0      7.1   1.3  38.5   
1     10.1   2.0  47.5   
2     13.0   2.3  42.3   
3     13.6   3.4  43.5   
4     20.8   6.9  35.8   
...    ...   ...   ...   
1977   9.4   1.8  54.1   
1978   5.2   2.7  40.4   
1979   9.0   3.6  42.9   
1980  26.7   3.7  40.8   
1981   1.9   5.5  42.2   

     dengue or probable (suspected) CABI(CAB) or confi

In [12]:
cbc_dataset_dengue.loc[cbc_dataset_dengue['clinical_diag'] == 'dengue', 'clinical_diag'] = 'Dengue'
cbc_dataset_dengue['clinical_diag'].unique()

array(['Dengue', 'Acute gastroenteritis', 'Meningococcus', 'UTI',
       'Pneumonia ', 'Typhoid fever', 'Skin infection', 'Leptospirosis',
       'Biliary tract infection/abdominal infection',
       'Others (infectious)', 'Diphtheria', 'CNS Infection', 'Bacteremia',
       'Unknown', 'Pneumonia', 'Enteric Fever',
       'Intraabdominal infection', 'Acute Pyelonephritis', 'septic rash'],
      dtype=object)

In [13]:
count_per_diag = cbc_dataset_dengue['clinical_diag'].value_counts()
print(count_per_diag)


clinical_diag
Others (infectious)                            672
Dengue                                         370
Pneumonia                                      195
Typhoid fever                                  132
Pneumonia                                      120
Leptospirosis                                  119
Skin infection                                 114
UTI                                             76
CNS Infection                                   52
Diphtheria                                      50
Meningococcus                                   31
Unknown                                         12
Intraabdominal infection                        12
Acute gastroenteritis                            9
Bacteremia                                       6
Biliary tract infection/abdominal infection      4
septic rash                                      4
Enteric Fever                                    2
Acute Pyelonephritis                             2
Name: count, dtyp

รคที่เกี่ยวข้องกับการติดเชื้อในกระแสเลือด โดยตรง มีดังนี้: <p>

Meningococcus: เยื่อหุ้มสมองอักเสบจากเชื้อเมนิงโกคอคคัส<p>
Bacteremia: ภาวะติดเชื้อในกระแสเลือด<p>
Septic rash: ผื่นติดเชื้อ<p>
โรคอื่นๆ อาจนำไปสู่ภาวะแทรกซ้อนที่รุนแรง เช่น การติดเชื้อในกระแสเลือด ได้แก่:<p>

UTI: การติดเชื้อในระบบทางเดินปัสสาวะ<p>
Typhoid fever: ไข้ไทฟอยด์<p>
Leptospirosis: โรคฉี่หนู<p>
Biliary tract infection/abdominal infection: การติดเชื้อในระบบทางเดินน้ำดี/การติดเชื้อในช่องท้อง<p>
Diphtheria: โรคคอตีบ<p>
Enteric Fever: ไข้รากสาดใหญ่<p>
Intraabdominal infection: การติดเชื้อในช่องท้อง<p>
Acute Pyelonephritis: กรวยไตอักเสบเฉียบพลัน<p>
Dengue<p>
Others (infectious) 

In [14]:
count_per_diag = cbc_dataset_dengue["clinical_diag"].value_counts()

diagnosis_list = [
    "Meningococcus",
    "Dengue",
    "Others (infectious)",
    "Bacteremia",
    "septic rash",
    "UTI",
    "Typhoid fever",
    "Leptospirosis",
    "Biliary tract infection/abdominal infection",
    "Diphtheria",
    "Enteric Fever",
    "Intraabdominal infection",
    "Acute Pyelonephritis",
]

for diagnosis in diagnosis_list:
    if diagnosis in count_per_diag.index:
        print("จำนวนของ", diagnosis, "คือ:", count_per_diag[diagnosis])
    else:
        print("ไม่พบข้อมูลสำหรับ", diagnosis)

จำนวนของ Meningococcus คือ: 31
จำนวนของ Dengue คือ: 370
จำนวนของ Others (infectious) คือ: 672
จำนวนของ Bacteremia คือ: 6
จำนวนของ septic rash คือ: 4
จำนวนของ UTI คือ: 76
จำนวนของ Typhoid fever คือ: 132
จำนวนของ Leptospirosis คือ: 119
จำนวนของ Biliary tract infection/abdominal infection คือ: 4
จำนวนของ Diphtheria คือ: 50
จำนวนของ Enteric Fever คือ: 2
จำนวนของ Intraabdominal infection คือ: 12
จำนวนของ Acute Pyelonephritis คือ: 2


In [15]:
filtered_data = cbc_dataset_dengue[cbc_dataset_dengue['clinical_diag'].isin(diagnosis_list)]

In [16]:
cbc_dataset_dengue = filtered_data
cbc_dataset_dengue

id  age     sex date of admission date of first symptom  \
0       1   21    male        2017-05-16            2017-05-11   
1       1   21    male        2017-05-16            2017-05-11   
2       1   21    male        2017-05-16            2017-05-11   
3       1   21    male        2017-05-16            2017-05-11   
4       1   21    male        2017-05-16            2017-05-11   
...   ...  ...     ...               ...                   ...   
1977  739   19    male        2018-09-24            2018-09-22   
1978  740   22    male        2018-09-24            2018-09-21   
1979  740   22    male        2018-09-24            2018-09-21   
1980  741   18    male        2018-09-26                   NaT   
1981  742   18  female        2018-09-26            2018-09-21   

            clinical_diag diagnosis  fatal_outcome  tbil    ast    alt    bun  \
0                  Dengue    dengue              0   NaN    NaN  122.0  10.67   
1                  Dengue    dengue              0   NaN    NaN  122.0  10.67   
2                  Dengue    dengue              0   NaN    NaN  122.0  10.67   
3                  Dengue    dengue              0   NaN    NaN  122.0  10.67   
4                  Dengue    dengue              0   NaN    NaN  122.0  10.67   
...                   ...       ...            ...   ...    ...    ...    ...   
1977  Others (infectious)      none              0   NaN  141.0   37.0  13.45   
1978  Others (infectious)      none              0   NaN  113.0   54.0   9.53   
1979  Others (infectious)      none              0   NaN  113.0   54.0   9.53   
1980  Others (infectious)      none              0   NaN    NaN    NaN    NaN   
1981  Others (infectious)      none              0   NaN    NaN    NaN    NaN   

      creatinin  crp  procalcitonin date of sampling sampling Time   wbc  \
0          0.63  0.6            1.7       2017-05-16      16:37:18  2.98   
1          0.63  0.6            1.7       2017-05-18      10:26:53  3.04   
2          0.63  0.6            1.7       2017-05-18      10:28:08  4.84   
3          0.63  0.6            1.7       2017-05-19      11:09:47  6.39   
4          0.63  0.6            1.7       2017-05-19      11:11:02  7.29   
...         ...  ...            ...              ...           ...   ...   
1977       0.90  0.6            0.1       2018-09-26      15:07:09  7.07   
1978       0.80  0.5            0.1       2018-09-25      11:29:46  1.79   
1979       0.80  0.5            0.1       2018-09-26      15:08:25  2.02   
1980        NaN  0.1            0.1       2018-09-26      14:06:16  9.01   
1981        NaN  0.6            0.1       2018-09-26      14:07:31  7.51   

      neut%  lymp%  mono%  eosino%  neut  lymp  mono  eosion   rbc   hb  plt  \
0      10.8   77.9    6.0      4.0  0.32  2.32  0.18    0.12  4.53  131   19   
1      20.0   63.2   11.2      3.3  0.61  1.92  0.34    0.10  4.74  136   20   
2      18.2   51.0   16.9     11.2  0.88  2.47  0.82    0.54  4.44  128   18   
3      20.6   45.2   12.8     19.7  1.31  2.89  0.82    1.26  4.53  130   25   
4      25.5   39.8    9.7     24.3  1.86  2.90  0.71    1.77  4.23  122   33   
...     ...    ...    ...      ...   ...   ...   ...     ...   ...  ...  ...   
1977   50.1   34.2   15.0      0.1  3.54  2.42  1.06    0.01  5.67  164   19   
1978   62.5   30.2    6.7      0.6  1.12  0.54  0.12    0.01  5.07  149   52   
1979   52.4   32.7   13.9      0.0  1.06  0.66  0.28    0.00  4.87  141   40   
1980   37.2   38.3   12.1     12.1  3.35  3.45  1.09    1.09  5.00  140   14   
1981   60.4   28.0    9.9      1.6  4.54  2.10  0.74    0.12  5.00  140  289   

      ipf%  ipf#   hct  \
0      7.1   1.3  38.5   
1     10.1   2.0  47.5   
2     13.0   2.3  42.3   
3     13.6   3.4  43.5   
4     20.8   6.9  35.8   
...    ...   ...   ...   
1977   9.4   1.8  54.1   
1978   5.2   2.7  40.4   
1979   9.0   3.6  42.9   
1980  26.7   3.7  40.8   
1981   1.9   5.5  42.2   

     dengue or probable (suspected) CABI(CAB) or confi

In [17]:
count_per_diag = cbc_dataset_dengue['clinical_diag'].value_counts()
print(count_per_diag)

clinical_diag
Others (infectious)                            672
Dengue                                         370
Typhoid fever                                  132
Leptospirosis                                  119
UTI                                             76
Diphtheria                                      50
Meningococcus                                   31
Intraabdominal infection                        12
Bacteremia                                       6
Biliary tract infection/abdominal infection      4
septic rash                                      4
Enteric Fever                                    2
Acute Pyelonephritis                             2
Name: count, dtype: int64


-------

In [18]:
cbc_dataset_meandeley.head()


index  S.No.   Age  Sex   RBC   PCV   MCV   MCH  MCHC   RDW    TLC  \
0    0.0    1.0  28.0  0.0  5.66  34.0  60.1  17.0  28.2  20.0  11.10   
1    1.0    2.0  41.0  0.0  4.78  44.5  93.1  28.9  31.0  13.0   7.02   
2    2.0    3.0  40.0  1.0  4.65  41.6  89.5  28.8  32.2  13.0   8.09   
3    3.0    4.0  76.0  0.0  4.24  36.7  86.6  26.7  30.8  14.9  13.41   
4    4.0    5.0  20.0  1.0  4.14  36.9  89.1  27.8  31.2  13.2   4.75   

   PLT/mm3   HGB  diseased  
0    128.3   9.6       1.0  
1    419.0  13.8       0.0  
2    325.0  13.4       0.0  
3    264.0  11.3       1.0  
4    196.0  11.5       0.0

In [19]:
# สร้าง DataFrame สำหรับค่าที่เป็นโรคโลหิตจาง (diseased = 1)
cbc_diseased_df = cbc_dataset_meandeley.groupby('diseased').get_group(1)

# สร้าง DataFrame สำหรับค่าที่ไม่เป็นโรคโลหิตจาง (diseased = 0)
cbc_non_diseased_df = cbc_dataset_meandeley.groupby('diseased').get_group(0)



In [20]:

# แสดงผลลัพธ์
print("DataFrame สำหรับค่าที่เป็นโรคโลหิตจาง:")
cbc_diseased_df.shape



DataFrame สำหรับค่าที่เป็นโรคโลหิตจาง:


(275, 14)

In [21]:

print("\nDataFrame สำหรับค่าที่ไม่เป็นโรคโลหิตจาง:")
cbc_non_diseased_df.shape



DataFrame สำหรับค่าที่ไม่เป็นโรคโลหิตจาง:


(63, 14)

-------------------

In [ ]:
cbc_dataset_kenya


-------

In [23]:
cbc_dataset_sav.columns


Index(['NO', 'GRUP', 'YAŞ', 'BMI', 'G', 'P', 'A', 'Y', 'GESTASYONELHAFTA',
       'WBC', 'HGB', 'HCT', 'sırı_microlitre', 'PIVI_microlitre', 'Sİİ', 'PLT',
       'NÖTROFİL', 'LENFOSİT', 'MONOSİT', 'ALT', 'AST', 'KREATİNİN', 'BUN',
       'DANSİTE', 'PLR', 'NLR', 'SIRI', 'PIV', 'notrofil_microlitre',
       'lenfosit_microlitre', 'platalet_microlitre', 'monosit_microlitre',
       'PUQE', 'PUQE_CID', 'PUQE_CID_İKİ'],
      dtype='object')

In [24]:
# สร้าง DataFrame สำหรับค่าที่ปกติ (nvp = 1)
cbc_sav_non_nvp_df = cbc_dataset_sav.groupby('GRUP').get_group(1)

# สร้าง DataFrame สำหรับค่าที่แพ้ท้อง (non-nvp = 0)
cbc_sav_nvp_df = cbc_dataset_sav.groupby('GRUP').get_group(0)

In [25]:
cbc_sav_non_nvp_df.shape

(278, 35)

-------

In [26]:
c = cbc_dataset_dengue.shape[0] + cbc_sav_non_nvp_df.shape[0] + cbc_dataset_kenya.shape[0] + cbc_diseased_df.shape[0] + cbc_non_diseased_df.shape[0]
c

2624

-------

Streptococcal septicemia <p>
Severe sepsis<p>
Sepsis<p>
Thrombocytopenia NOS<p>
Anemia NOS<p>
Gram-neg septicemia NOS<p>
Septicemia NOS<p>
Thrombocytopenia NOS<p>
Hemorrhage complic proc<p>

---
oter prepare Data<p>
---

In [27]:
cbc_dataset_dengue

id  age     sex date of admission date of first symptom  \
0       1   21    male        2017-05-16            2017-05-11   
1       1   21    male        2017-05-16            2017-05-11   
2       1   21    male        2017-05-16            2017-05-11   
3       1   21    male        2017-05-16            2017-05-11   
4       1   21    male        2017-05-16            2017-05-11   
...   ...  ...     ...               ...                   ...   
1977  739   19    male        2018-09-24            2018-09-22   
1978  740   22    male        2018-09-24            2018-09-21   
1979  740   22    male        2018-09-24            2018-09-21   
1980  741   18    male        2018-09-26                   NaT   
1981  742   18  female        2018-09-26            2018-09-21   

            clinical_diag diagnosis  fatal_outcome  tbil    ast    alt    bun  \
0                  Dengue    dengue              0   NaN    NaN  122.0  10.67   
1                  Dengue    dengue              0   NaN    NaN  122.0  10.67   
2                  Dengue    dengue              0   NaN    NaN  122.0  10.67   
3                  Dengue    dengue              0   NaN    NaN  122.0  10.67   
4                  Dengue    dengue              0   NaN    NaN  122.0  10.67   
...                   ...       ...            ...   ...    ...    ...    ...   
1977  Others (infectious)      none              0   NaN  141.0   37.0  13.45   
1978  Others (infectious)      none              0   NaN  113.0   54.0   9.53   
1979  Others (infectious)      none              0   NaN  113.0   54.0   9.53   
1980  Others (infectious)      none              0   NaN    NaN    NaN    NaN   
1981  Others (infectious)      none              0   NaN    NaN    NaN    NaN   

      creatinin  crp  procalcitonin date of sampling sampling Time   wbc  \
0          0.63  0.6            1.7       2017-05-16      16:37:18  2.98   
1          0.63  0.6            1.7       2017-05-18      10:26:53  3.04   
2          0.63  0.6            1.7       2017-05-18      10:28:08  4.84   
3          0.63  0.6            1.7       2017-05-19      11:09:47  6.39   
4          0.63  0.6            1.7       2017-05-19      11:11:02  7.29   
...         ...  ...            ...              ...           ...   ...   
1977       0.90  0.6            0.1       2018-09-26      15:07:09  7.07   
1978       0.80  0.5            0.1       2018-09-25      11:29:46  1.79   
1979       0.80  0.5            0.1       2018-09-26      15:08:25  2.02   
1980        NaN  0.1            0.1       2018-09-26      14:06:16  9.01   
1981        NaN  0.6            0.1       2018-09-26      14:07:31  7.51   

      neut%  lymp%  mono%  eosino%  neut  lymp  mono  eosion   rbc   hb  plt  \
0      10.8   77.9    6.0      4.0  0.32  2.32  0.18    0.12  4.53  131   19   
1      20.0   63.2   11.2      3.3  0.61  1.92  0.34    0.10  4.74  136   20   
2      18.2   51.0   16.9     11.2  0.88  2.47  0.82    0.54  4.44  128   18   
3      20.6   45.2   12.8     19.7  1.31  2.89  0.82    1.26  4.53  130   25   
4      25.5   39.8    9.7     24.3  1.86  2.90  0.71    1.77  4.23  122   33   
...     ...    ...    ...      ...   ...   ...   ...     ...   ...  ...  ...   
1977   50.1   34.2   15.0      0.1  3.54  2.42  1.06    0.01  5.67  164   19   
1978   62.5   30.2    6.7      0.6  1.12  0.54  0.12    0.01  5.07  149   52   
1979   52.4   32.7   13.9      0.0  1.06  0.66  0.28    0.00  4.87  141   40   
1980   37.2   38.3   12.1     12.1  3.35  3.45  1.09    1.09  5.00  140   14   
1981   60.4   28.0    9.9      1.6  4.54  2.10  0.74    0.12  5.00  140  289   

      ipf%  ipf#   hct  \
0      7.1   1.3  38.5   
1     10.1   2.0  47.5   
2     13.0   2.3  42.3   
3     13.6   3.4  43.5   
4     20.8   6.9  35.8   
...    ...   ...   ...   
1977   9.4   1.8  54.1   
1978   5.2   2.7  40.4   
1979   9.0   3.6  42.9   
1980  26.7   3.7  40.8   
1981   1.9   5.5  42.2   

     dengue or probable (suspected) CABI(CAB) or confi

-------define-data----------

<h3>Must Have</h3><p>
Age<p>
Gender<p>
Red cell count (RBC) <p>
Hemoglobin (Hb) <p>
Hematocrit (Hct) <p>
Mean corpuscular volume (MCV) <p>
Mean corpuscular hemoglobin (MCH) <p>
Mean corpuscular hemoglobin concentration (MCHC) <p>
Red cell distribution width (RDW) <p>
Total Lymphocyte Count (TLC ) <p>
White Blood Cell Count (WBC) <p>
การแข็งตัวของเลือด (Platelets)

In [28]:
cbc_dataset_dengue

id  age     sex date of admission date of first symptom  \
0       1   21    male        2017-05-16            2017-05-11   
1       1   21    male        2017-05-16            2017-05-11   
2       1   21    male        2017-05-16            2017-05-11   
3       1   21    male        2017-05-16            2017-05-11   
4       1   21    male        2017-05-16            2017-05-11   
...   ...  ...     ...               ...                   ...   
1977  739   19    male        2018-09-24            2018-09-22   
1978  740   22    male        2018-09-24            2018-09-21   
1979  740   22    male        2018-09-24            2018-09-21   
1980  741   18    male        2018-09-26                   NaT   
1981  742   18  female        2018-09-26            2018-09-21   

            clinical_diag diagnosis  fatal_outcome  tbil    ast    alt    bun  \
0                  Dengue    dengue              0   NaN    NaN  122.0  10.67   
1                  Dengue    dengue              0   NaN    NaN  122.0  10.67   
2                  Dengue    dengue              0   NaN    NaN  122.0  10.67   
3                  Dengue    dengue              0   NaN    NaN  122.0  10.67   
4                  Dengue    dengue              0   NaN    NaN  122.0  10.67   
...                   ...       ...            ...   ...    ...    ...    ...   
1977  Others (infectious)      none              0   NaN  141.0   37.0  13.45   
1978  Others (infectious)      none              0   NaN  113.0   54.0   9.53   
1979  Others (infectious)      none              0   NaN  113.0   54.0   9.53   
1980  Others (infectious)      none              0   NaN    NaN    NaN    NaN   
1981  Others (infectious)      none              0   NaN    NaN    NaN    NaN   

      creatinin  crp  procalcitonin date of sampling sampling Time   wbc  \
0          0.63  0.6            1.7       2017-05-16      16:37:18  2.98   
1          0.63  0.6            1.7       2017-05-18      10:26:53  3.04   
2          0.63  0.6            1.7       2017-05-18      10:28:08  4.84   
3          0.63  0.6            1.7       2017-05-19      11:09:47  6.39   
4          0.63  0.6            1.7       2017-05-19      11:11:02  7.29   
...         ...  ...            ...              ...           ...   ...   
1977       0.90  0.6            0.1       2018-09-26      15:07:09  7.07   
1978       0.80  0.5            0.1       2018-09-25      11:29:46  1.79   
1979       0.80  0.5            0.1       2018-09-26      15:08:25  2.02   
1980        NaN  0.1            0.1       2018-09-26      14:06:16  9.01   
1981        NaN  0.6            0.1       2018-09-26      14:07:31  7.51   

      neut%  lymp%  mono%  eosino%  neut  lymp  mono  eosion   rbc   hb  plt  \
0      10.8   77.9    6.0      4.0  0.32  2.32  0.18    0.12  4.53  131   19   
1      20.0   63.2   11.2      3.3  0.61  1.92  0.34    0.10  4.74  136   20   
2      18.2   51.0   16.9     11.2  0.88  2.47  0.82    0.54  4.44  128   18   
3      20.6   45.2   12.8     19.7  1.31  2.89  0.82    1.26  4.53  130   25   
4      25.5   39.8    9.7     24.3  1.86  2.90  0.71    1.77  4.23  122   33   
...     ...    ...    ...      ...   ...   ...   ...     ...   ...  ...  ...   
1977   50.1   34.2   15.0      0.1  3.54  2.42  1.06    0.01  5.67  164   19   
1978   62.5   30.2    6.7      0.6  1.12  0.54  0.12    0.01  5.07  149   52   
1979   52.4   32.7   13.9      0.0  1.06  0.66  0.28    0.00  4.87  141   40   
1980   37.2   38.3   12.1     12.1  3.35  3.45  1.09    1.09  5.00  140   14   
1981   60.4   28.0    9.9      1.6  4.54  2.10  0.74    0.12  5.00  140  289   

      ipf%  ipf#   hct  \
0      7.1   1.3  38.5   
1     10.1   2.0  47.5   
2     13.0   2.3  42.3   
3     13.6   3.4  43.5   
4     20.8   6.9  35.8   
...    ...   ...   ...   
1977   9.4   1.8  54.1   
1978   5.2   2.7  40.4   
1979   9.0   3.6  42.9   
1980  26.7   3.7  40.8   
1981   1.9   5.5  42.2   

     dengue or probable (suspected) CABI(CAB) or confi

In [29]:
cbc_dataset_dengue.columns

Index(['id', 'age', 'sex', 'date of admission', 'date of first symptom',
       'clinical_diag', 'diagnosis', 'fatal_outcome', 'tbil', 'ast', 'alt',
       'bun', 'creatinin', 'crp', 'procalcitonin', 'date of sampling',
       'sampling Time', 'wbc', 'neut%', 'lymp%', 'mono%', 'eosino%', 'neut',
       'lymp', 'mono', 'eosion', 'rbc', 'hb', 'plt', 'ipf%', 'ipf#', 'hct',
       'dengue or probable (suspected) CABI(CAB) or confirmed CABI(CAB)',
       'severe_dengue'],
      dtype='object')

In [30]:
key_word_drop = [
    "clinical_diag",
    "diagnosis",
    "id",
    'creatinin',
    "fatal_outcome",
    "ast",
    "alt",
    "bun",
    "tbil",
    "procalcitonin",
    "ipf%",
    "ipf#",
    "eosino%",
    "neut",
    "rbc",
    "lymp",
    "crp",
    "lymp%",
    "eosion",
    "mono",
    "date of admission",
    "date of first symptom",
    "date of sampling",
    "sampling Time",
    "severe_dengue",
]
new_cbc_dataset_dengue = cbc_dataset_dengue.drop(columns=key_word_drop)

In [31]:
new_cbc_dataset_dengue

age     sex   wbc  neut%  mono%   hb  plt   hct  \
0      21    male  2.98   10.8    6.0  131   19  38.5   
1      21    male  3.04   20.0   11.2  136   20  47.5   
2      21    male  4.84   18.2   16.9  128   18  42.3   
3      21    male  6.39   20.6   12.8  130   25  43.5   
4      21    male  7.29   25.5    9.7  122   33  35.8   
...   ...     ...   ...    ...    ...  ...  ...   ...   
1977   19    male  7.07   50.1   15.0  164   19  54.1   
1978   22    male  1.79   62.5    6.7  149   52  40.4   
1979   22    male  2.02   52.4   13.9  141   40  42.9   
1980   18    male  9.01   37.2   12.1  140   14  40.8   
1981   18  female  7.51   60.4    9.9  140  289  42.2   

     dengue or probable (suspected) CABI(CAB) or confirmed CABI(CAB)  
0                                                dengue               
1                                                dengue               
2                                                dengue               
3                                                dengue               
4                                                dengue               
...                                                 ...               
1977                                       probable CAB               
1978                                       probable CAB               
1979                                       probable CAB               
1980                                       probable CAB               
1981                                       probable CAB               

[1480 rows x 9 columns]

In [32]:
new_cbc_dataset_dengue.rename(columns={'dengue or probable (suspected) CABI(CAB) or confirmed CABI(CAB)': 'Diagnosis'}, inplace=True)

In [33]:
new_cbc_dataset_dengue

age     sex   wbc  neut%  mono%   hb  plt   hct     Diagnosis
0      21    male  2.98   10.8    6.0  131   19  38.5        dengue
1      21    male  3.04   20.0   11.2  136   20  47.5        dengue
2      21    male  4.84   18.2   16.9  128   18  42.3        dengue
3      21    male  6.39   20.6   12.8  130   25  43.5        dengue
4      21    male  7.29   25.5    9.7  122   33  35.8        dengue
...   ...     ...   ...    ...    ...  ...  ...   ...           ...
1977   19    male  7.07   50.1   15.0  164   19  54.1  probable CAB
1978   22    male  1.79   62.5    6.7  149   52  40.4  probable CAB
1979   22    male  2.02   52.4   13.9  141   40  42.9  probable CAB
1980   18    male  9.01   37.2   12.1  140   14  40.8  probable CAB
1981   18  female  7.51   60.4    9.9  140  289  42.2  probable CAB

[1480 rows x 9 columns]

In [34]:
sum_of_nans = new_cbc_dataset_dengue.isna().sum()
print(sum_of_nans)

age          0
sex          0
wbc          0
neut%        0
mono%        0
hb           0
plt          0
hct          0
Diagnosis    0
dtype: int64


-----

In [35]:
cbc_sav_non_nvp_df

NO  GRUP   YAŞ   BMI    G    P    A    Y  GESTASYONELHAFTA      WBC  \
278  279.0   1.0  23.0  26.0  1.0  0.0  0.0  0.0               9.1   7480.0   
279  280.0   1.0  27.0  26.0  2.0  1.0  0.0  1.0               9.1   6780.0   
280  281.0   1.0  25.0  28.0  1.0  0.0  0.0  0.0               6.0   9710.0   
281  282.0   1.0  33.0  24.0  4.0  2.0  1.0  2.0               7.0   9370.0   
282  283.0   1.0  24.0  20.0  1.0  0.0  0.0  0.0               9.1  11430.0   
283  284.0   1.0  19.0  23.0  1.0  0.0  0.0  0.0              14.3  10450.0   
284  285.0   1.0  28.0  25.0  1.0  0.0  0.0  0.0               9.6   9940.0   
285  286.0   1.0  28.0  25.0  1.0  0.0  0.0  0.0               9.2   8120.0   
286  287.0   1.0  27.0  36.0  2.0  1.0  0.0  1.0               9.5   8790.0   
287  288.0   1.0  33.0  36.0  3.0  1.0  1.0  1.0              10.4   9910.0   
288  289.0   1.0  29.0  23.0  2.0  1.0  0.0  1.0              10.2  11650.0   
289  290.0   1.0  20.0  39.0  1.0  0.0  0.0  0.0              12.0   6320.0   
290  291.0   1.0  29.0  25.0  1.0  0.0  0.0  0.0               8.6   8860.0   
291  292.0   1.0  27.0  21.0  3.0  1.0  1.0  1.0              12.0   8920.0   
292  293.0   1.0  26.0  29.0  1.0  0.0  0.0  0.0               7.0   6390.0   
293  294.0   1.0  30.0  27.0  2.0  1.0  0.0  1.0               6.1   8270.0   
294  295.0   1.0  22.0  26.0  1.0  0.0  0.0  0.0              11.6   8050.0   
295  296.0   1.0  25.0  20.0  1.0  0.0  0.0  0.0               8.5   6440.0   
296  297.0   1.0  32.0  33.0  2.0  1.0  0.0  1.0              10.2  12770.0   
297  298.0   1.0  25.0  24.0  2.0  0.0  1.0  0.0              10.0   8400.0   
298  299.0   1.0  30.0  21.0  2.0  1.0  0.0  1.0              12.0  13280.0   
299  300.0   1.0  29.0  30.0  1.0  0.0  0.0  0.0               7.5   9560.0   
300  301.0   1.0  31.0  21.0  1.0  0.0  0.0  0.0               8.4  10970.0   
301  302.0   1.0  29.0  27.0  2.0  1.0  0.0  1.0              11.2  10760.0   
302  303.0   1.0  27.0  24.0  1.0  0.0  0.0  0.0               7.5   8410.0   
303  304.0   1.0  20.0  21.0  1.0  0.0  0.0  0.0               7.0  10910.0   
304  305.0   1.0  30.0  30.0  3.0  2.0  0.0  2.0               7.4   7890.0   
305  306.0   1.0  26.0  26.0  1.0  0.0  0.0  0.0              11.0  10390.0   
306  307.0   1.0  24.0  33.0  1.0  0.0  0.0  0.0               9.6  11010.0   
307  308.0   1.0  25.0  21.0  1.0  0.0  0.0  0.0               9.0  14760.0   
308  309.0   1.0  29.0  20.0  2.0  1.0  0.0  1.0               9.0   9440.0   
309  310.0   1.0  23.0  24.0  2.0  0.0  1.0  0.0               9.4  11200.0   
310  311.0   1.0  25.0  27.0  1.0  0.0  0.0  0.0               9.2   5930.0   
311  312.0   1.0  25.0  20.0  1.0  0.0  0.0  0.0               9.0   5810.0   
312  313.0   1.0  36.0  33.0  2.0  0.0  1.0  0.0              11.1  10930.0   
313  314.0   1.0  20.0  21.0  2.0  0.0  1.0  0.0               6.5   6010.0   
314  315.0   1.0  38.0  37.0  3.0  2.0  0.0  2.0               8.5   5080.0   
315  316.0   1.0  31.0  33.0  3.0  1.0  1.0  1.0               7.5  12380.0   
316  317.0   1.0  26.0  21.0  1.0  0.0  0.0  0.0               9.2   7460.0   
317  318.0   1.0  39.0  23.0  3.0  2.0  0.0  2.0              13.4  10340.0   
318  319.0   1.0  33.0  30.0  3.0  0.0  2.0  0.0               9.3  11030.0   
319  320.0   1.0  37.0  21.0  3.0  2.0  0.0  2.0              12.1   3910.0   
320  321.0   1.0  31.0  32.0  1.0  0.0  0.0  0.0               6.0   7560.0   
321  322.0   1.0  26.0  19.0  1.0  0.0  0.0  0.0              13.3   6910.0   
322  323.0   1.0  25.0  26.0  1.0  0.0  0.0  0.0              11.1   7950.0   
323  324.0   1.0  38.0  23.0  3.0  2.0  0.0  2.0               8.1  11320.0   
324  325.0   1.0  28.0  22.0  1.0  0.0  0.0  0.0              13.5   9510.0   
325  326.0   1.0  24.0  22.0  1.0  0.0  0.0  0.0               6.2   9440.0   
326  327.0   1.0  25.0  33.0  1.0  0.0  0.0  0.0              15.6  11370.0   
327  328.0   1.0  28.0  30.0  1.0  0.0  0.0  0.0          

In [36]:
cbc_sav_non_nvp_df.head()

NO  GRUP   YAŞ   BMI    G    P    A    Y  GESTASYONELHAFTA      WBC  \
278  279.0   1.0  23.0  26.0  1.0  0.0  0.0  0.0               9.1   7480.0   
279  280.0   1.0  27.0  26.0  2.0  1.0  0.0  1.0               9.1   6780.0   
280  281.0   1.0  25.0  28.0  1.0  0.0  0.0  0.0               6.0   9710.0   
281  282.0   1.0  33.0  24.0  4.0  2.0  1.0  2.0               7.0   9370.0   
282  283.0   1.0  24.0  20.0  1.0  0.0  0.0  0.0               9.1  11430.0   

      HGB   HCT  sırı_microlitre  PIVI_microlitre           Sİİ       PLT  \
278  13.1  39.0         1.903846       670.153846  1.489231e+06  352000.0   
279  14.8  45.4         2.045455       613.636364  1.363636e+06  300000.0   
280  13.6  40.0         4.646667       989.740000  1.940667e+06  213000.0   
281  13.0  37.0         1.857143       776.285714  1.293810e+06  418000.0   
282  13.2  40.0         2.048000       649.216000  1.014400e+06  317000.0   

     NÖTROFİL  LENFOSİT  MONOSİT   ALT   AST  KREATİNİN   BUN  DANSİTE  \
278    5500.0    1300.0    450.0  13.0  18.0        0.6   7.0   1025.0   
279    5000.0    1100.0    450.0  12.0  15.0        0.4   9.0   1020.0   
280    8200.0     900.0    510.0  13.0  15.0        0.4   7.0   1015.0   
281    6500.0    2100.0    600.0   6.0  15.0        0.6   8.0   1030.0   
282    8000.0    2500.0    640.0  13.0  17.0        0.4  10.0   1030.0   

            PLR       NLR         SIRI           PIV  notrofil_microlitre  \
278  270.769231  4.230769  1903.846154  6.701538e+08                  5.5   
279  272.727273  4.545455  2045.454545  6.136364e+08                  5.0   
280  236.666667  9.111111  4646.666667  9.897400e+08                  8.2   
281  199.047619  3.095238  1857.142857  7.762857e+08                  6.5   
282  126.800000  3.200000  2048.000000  6.492160e+08                  8.0   

     lenfosit_microlitre  platalet_microlitre  monosit_microlitre  PUQE  \
278                  1.3                352.0                0.45   2.0   
279                  1.1                300.0                0.45   3.0   
280                  0.9                213.0                0.51   2.0   
281                  2.1                418.0                0.60   2.0   
282                  2.5                317.0                0.64   3.0   

     PUQE_CID  PUQE_CID_İKİ  
278       0.0           1.0  
279       1.0           1.0  
280       0.0           1.0  
281       0.0           1.0  
282       1.0           1.0

In [37]:
cbc_sav_non_nvp_df.columns

Index(['NO', 'GRUP', 'YAŞ', 'BMI', 'G', 'P', 'A', 'Y', 'GESTASYONELHAFTA',
       'WBC', 'HGB', 'HCT', 'sırı_microlitre', 'PIVI_microlitre', 'Sİİ', 'PLT',
       'NÖTROFİL', 'LENFOSİT', 'MONOSİT', 'ALT', 'AST', 'KREATİNİN', 'BUN',
       'DANSİTE', 'PLR', 'NLR', 'SIRI', 'PIV', 'notrofil_microlitre',
       'lenfosit_microlitre', 'platalet_microlitre', 'monosit_microlitre',
       'PUQE', 'PUQE_CID', 'PUQE_CID_İKİ'],
      dtype='object')

In [38]:
cbc_sav_non_nvp_df = cbc_sav_non_nvp_df.reset_index()

In [39]:
# เลือกและเก็บเฉพาะคอลัมน์ที่ต้องการจาก DataFrame ใหม่
selected_columns_df = cbc_sav_non_nvp_df[['YAŞ','HGB','HCT','WBC','PLT','NÖTROFİL','MONOSİT']]
# สร้างคอลัมน์ใหม่และกำหนดค่าเดียวเพียงค่าเดียวให้กับทุกแถว
selected_columns_df['sex'] = 'female'
selected_columns_df['Diagnosis'] = 'normal'
new_cbc_sav_non_nvp_df = selected_columns_df






C:\Users\Bravo\AppData\Local\Temp\ipykernel_17800\3124824626.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  selected_columns_df['sex'] = 'female'
C:\Users\Bravo\AppData\Local\Temp\ipykernel_17800\3124824626.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  selected_columns_df['Diagnosis'] = 'normal'


In [40]:
new_cbc_sav_non_nvp_df.head()

YAŞ   HGB   HCT      WBC       PLT  NÖTROFİL  MONOSİT     sex Diagnosis
0  23.0  13.1  39.0   7480.0  352000.0    5500.0    450.0  female    normal
1  27.0  14.8  45.4   6780.0  300000.0    5000.0    450.0  female    normal
2  25.0  13.6  40.0   9710.0  213000.0    8200.0    510.0  female    normal
3  33.0  13.0  37.0   9370.0  418000.0    6500.0    600.0  female    normal
4  24.0  13.2  40.0  11430.0  317000.0    8000.0    640.0  female    normal

In [41]:
sum_of_nans = new_cbc_sav_non_nvp_df.isna().sum()
print(sum_of_nans)

YAŞ          0
HGB          0
HCT          0
WBC          0
PLT          0
NÖTROFİL     0
MONOSİT      0
sex          0
Diagnosis    0
dtype: int64


-------

In [42]:
cbc_dataset_kenya

Age  Sex  Red cell count    Hb   Hct  MCV  MCH  MCHC   RDW    WBC  \
0     23    1            5.08  15.9  0.47   92   31    34  11.4   6.16   
1     24    0            5.61  17.3  0.51   90   31    34  12.7   6.67   
2     22    1            5.11  16.2  0.47   93   32    34  12.8   5.13   
3     22    0            5.54  16.3  0.48   86   29    34  14.1   3.10   
4     24    1            4.50  13.9  0.42   93   31    33  11.7   4.22   
5     21    0            5.46  16.6  0.49   90   31    34  13.1   3.30   
6     22    1            5.50  15.4  0.47   85   28    33  12.2   5.12   
7     23    0            5.70  17.7  0.52   91   31    34  12.4   5.10   
8     22    1            4.59  13.5  0.40   87   29    34  14.7   5.64   
9     23    0            6.53  17.5  0.53   81   27    33  14.2   4.30   
10    20    1            4.74  14.9  0.44   93   31    34  12.4   3.85   
11    22    0            5.08  15.5  0.46   91   31    33  13.4   5.85   
12    20    1            5.14  15.4  0.45   88   30    34  12.3   5.36   
13    23    0            5.99  16.8  0.49   82   28    34  14.0   6.48   
14    22    1            4.68  14.2  0.42   91   30    34  12.4  14.84   
15    25    0            6.06  16.2  0.49   81   27    33  14.6   5.28   
16    24    1            6.53  12.9  0.42   65   20    30  19.3   5.30   
17    40    0            5.80  15.5  0.49   85   27    32  12.4   4.84   
18    21    1            5.06  14.2  0.44   86   28    33  12.9   5.92   
19    22    0            5.95  16.7  0.51   85   28    33  14.6   6.55   
20    22    1            5.31  12.1  0.37   70   23    32  19.2   6.93   
21    23    0            5.38  15.8  0.48   88   29    33  13.1   3.50   
22    40    1            5.03  14.9  0.45   89   29    33  12.6   4.98   
23    23    0            5.38  16.9  0.51   95   31    33  10.8   7.04   
24    25    1            5.06  18.8  0.47   92   31    34  12.0   5.40   
25    23    0            5.49  17.9  0.53   96   33    34  11.7   4.73   
26    20    1            5.34  14.5  0.43   80   27    34  13.6   4.70   
27    24    0            5.98  18.1  0.53   89   30    34  13.6   5.31   
28    22    1            5.73  15.7  0.47   82   27    34  12.8   4.90   
29    24    0            6.05  17.2  0.51   84   28    34  13.1   6.74   
30    20    1            5.87  13.1  0.40   69   22    33  17.3   4.60   
31    25    0            5.90  18.4  0.55   93   31    34  13.2   5.10   
32    23    1            5.91  16.8  0.52   87   29    33  11.9   7.40   
33    38    0            5.49  15.1  0.45   81   28    34  13.5   4.91   
34    20    1            4.99  14.1  0.42   84   28    33  15.4   5.90   
35    34    0            5.20  16.9  0.48   91   33    36  13.3   5.67   
36    20    1            4.46  13.8  0.41   91   31    34  11.7   4.10   
37    23    0            6.08  15.7  0.48   80   26    32  14.4   6.69   
38    21    1            4.46  13.4  0.40   89   30    34  12.2   4.78   
39    20    0            5.66  16.7  0.50   88   30    34  13.0   6.55   
40    21    1            4.87  13.1  0.39   80   27    34  14.3   7.10   
41    27    0            6.04  17.4  0.51   85   29    34  13.4   5.83   
42    21    1            5.26  15.2  0.45   86   29    34  12.2   3.88   
43    25    0            5.18  15.4  0.45   87   30    34  12.8   3.91   
44    22    1            4.80  14.5  0.43   89   30    34  12.8   5.60   
45    41    0            5.34  14.9  0.45   84   28    33  11.8   3.24   
46    21    1            4.80  14.2  0.42   88   30    34  12.6   4.34   
47    24    0            5.57  15.5  0.46   83   28    34  13.4   5.65   
48    22    0            5.31  16.3  0.48   90   31    34  12.7   6.61   
49    21    1            5.40  16.5  0.48   90   31    34  11.8   3.09   
50    23    0            5.46  15.9  0.48   87   29    33  13.4   5.11   
51    18    1            4.68  15.0  0.44   94   32    34  12.6   3.88   
52    28    0            5.49  16.2  0.45   82   29    36  13.4   5.36   
53    20 

In [43]:
cbc_dataset_kenya.head()

Age  Sex  Red cell count    Hb   Hct  MCV  MCH  MCHC   RDW   WBC  \
0   23    1            5.08  15.9  0.47   92   31    34  11.4  6.16   
1   24    0            5.61  17.3  0.51   90   31    34  12.7  6.67   
2   22    1            5.11  16.2  0.47   93   32    34  12.8  5.13   
3   22    0            5.54  16.3  0.48   86   29    34  14.1  3.10   
4   24    1            4.50  13.9  0.42   93   31    33  11.7  4.22   

   Neutrophil %  Lymphocyte %  Monocyte %  Eosinophil %  Basophil %  \
0          48.6          39.7         6.7           4.3         0.7   
1          36.7          48.7        11.7           2.3         0.6   
2          36.0          54.0         8.4           1.0         0.6   
3          32.6          54.7        10.5           1.5         0.7   
4          50.3          35.7        12.0           1.4         0.6   

   Neutrophil Abs  Lymphocyte Abs  Monocyte Abs  Eosinophil Abs  Basophil Abs  \
0            2.99            2.45          0.41            0.26          0.04   
1            2.45            3.25          0.78            0.15          0.04   
2            1.85            2.77          0.43            0.05          0.03   
3            1.01            1.70          0.33            0.05          0.02   
4            2.12            1.51          0.51            0.06          0.03   

   Platelets  
0        198  
1        287  
2        336  
3        223  
4        367

 ---------------


In [44]:
cbc_dataset_kenya.columns

Index(['Age', 'Sex', 'Red cell count', 'Hb', 'Hct', 'MCV', 'MCH', 'MCHC',
       'RDW', 'WBC', 'Neutrophil %', 'Lymphocyte %', 'Monocyte %',
       'Eosinophil %', 'Basophil %', 'Neutrophil Abs', 'Lymphocyte Abs',
       'Monocyte Abs', 'Eosinophil Abs', 'Basophil Abs', 'Platelets'],
      dtype='object')

In [45]:
# เลือกและเก็บเฉพาะคอลัมน์ที่ต้องการจาก DataFrame ใหม่
selected_columns_df = cbc_dataset_kenya[['Age','Sex','Hb','Hct','WBC','Platelets','Neutrophil %','Monocyte %']]
new_cbc_dataset_kenya = selected_columns_df
new_cbc_dataset_kenya['Diagnosis'] = 'normal'


C:\Users\Bravo\AppData\Local\Temp\ipykernel_17800\3092057935.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  new_cbc_dataset_kenya['Diagnosis'] = 'normal'


In [46]:
new_cbc_dataset_kenya.head()

Age  Sex    Hb   Hct   WBC  Platelets  Neutrophil %  Monocyte % Diagnosis
0   23    1  15.9  0.47  6.16        198          48.6         6.7    normal
1   24    0  17.3  0.51  6.67        287          36.7        11.7    normal
2   22    1  16.2  0.47  5.13        336          36.0         8.4    normal
3   22    0  16.3  0.48  3.10        223          32.6        10.5    normal
4   24    1  13.9  0.42  4.22        367          50.3        12.0    normal

In [47]:
sum_of_nans = new_cbc_dataset_kenya.isna().sum()
print(sum_of_nans)

Age             0
Sex             0
Hb              0
Hct             0
WBC             0
Platelets       0
Neutrophil %    0
Monocyte %      0
Diagnosis       0
dtype: int64


 ---------------


In [48]:
cbc_dataset_diagnoses.head()

subject_id  Hemoglobin  Eosinophils  Lymphocytes  Monocytes  Basophils  \
0       10006         NaN          NaN          NaN        NaN        0.3   
1       10006         NaN          NaN          NaN        NaN        0.3   
2       10006         NaN          NaN          NaN        NaN        0.3   
3       10006         NaN          NaN          NaN        NaN        0.3   
4       10006         NaN          NaN          NaN        NaN        0.3   

   Eosinophils.1  Hematocrit  Hemoglobin.1  Lymphocytes.1        MCH  \
0           1.95   33.811111     11.133333          15.85  31.188889   
1           1.95   33.811111     11.133333          15.85  31.188889   
2           1.95   33.811111     11.133333          15.85  31.188889   
3           1.95   33.811111     11.133333          15.85  31.188889   
4           1.95   33.811111     11.133333          15.85  31.188889   

        MCHC        MCV  Monocytes.1  Neutrophils   RDW  Red Blood Cells  \
0  33.066667  94.444444         6.55         75.4  19.5         3.577778   
1  33.066667  94.444444         6.55         75.4  19.5         3.577778   
2  33.066667  94.444444         6.55         75.4  19.5         3.577778   
3  33.066667  94.444444         6.55         75.4  19.5         3.577778   
4  33.066667  94.444444         6.55         75.4  19.5         3.577778   

   White Blood Cells  Monocytes.2  Lymphocytes.2  Monocytes.3  Eosinophils.2  \
0           3.811111          NaN            NaN          NaN            NaN   
1           3.811111          NaN            NaN          NaN            NaN   
2           3.811111          NaN            NaN          NaN            NaN   
3           3.811111          NaN            NaN          NaN            NaN   
4           3.811111          NaN            NaN          NaN            NaN   

   Lymphocytes.3  Eosinophils.3  Lymphocytes.4 gender                  dob  \
0            NaN            NaN            NaN      F  2094-03-05 00:00:00   
1            NaN            NaN            NaN      F  2094-03-05 00:00:00   
2            NaN            NaN            NaN      F  2094-03-05 00:00:00   
3            NaN            NaN            NaN      F  2094-03-05 00:00:00   
4            NaN            NaN            NaN      F  2094-03-05 00:00:00   

                   dod             dod_hosp              dod_ssn  expire_flag  \
0  2165-08-12 00:00:00  2165-08-12 00:00:00  2165-08-12 00:00:00            1   
1  2165-08-12 00:00:00  2165-08-12 00:00:00  2165-08-12 00:00:00            1   
2  2165-08-12 00:00:00  2165-08-12 00:00:00  2165-08-12 00:00:00            1   
3  2165-08-12 00:00:00  2165-08-12 00:00:00  2165-08-12 00:00:00            1   
4  2165-08-12 00:00:00  2165-08-12 00:00:00  2165-08-12 00:00:00            1   

                short_title                                         long_title  
0                    Sepsis                                             Sepsis  
1  React-oth vasc dev/graft  Infection and inflammatory reaction due to oth...  
2    Hyp kid NOS w cr kid V  Hypertensive chronic kidney disease, unspecifi...  
3       Atrial fibrillation                                Atrial fibrillation  
4                   CHF NOS              Congestive heart failure, unspecified

In [49]:
keywords = ['Anemia NOS']

Anemia_cbc_dataset_diagnoses = cbc_dataset_diagnoses[cbc_dataset_diagnoses['short_title'].isin(keywords)]

# แสดงผลลัพธ์
Anemia_cbc_dataset_diagnoses.head()



subject_id  Hemoglobin  Eosinophils  Lymphocytes  Monocytes  Basophils  \
65        10026         NaN          NaN          NaN        NaN   0.600000   
199       10045   10.933333          NaN          NaN        NaN   0.000000   
343       10088    9.677778          NaN          NaN        NaN   0.121176   
462       10114         NaN          NaN          NaN        NaN   0.300000   
580       40124         NaN          NaN          NaN        NaN   0.225000   

     Eosinophils.1  Hematocrit  Hemoglobin.1  Lymphocytes.1        MCH  \
65        3.150000   27.975000      9.350000      21.750000  30.775000   
199       0.500000   31.466667     10.782609       1.500000  29.854545   
343       0.325882   30.575039     10.435273       9.421765  29.930000   
462       0.700000   30.677778     10.477778       9.000000  28.744444   
580       2.800000   26.586667      8.553333       4.675000  30.100000   

          MCHC        MCV  Monocytes.1  Neutrophils        RDW  \
65   33.450000  92.000000     6.050000    68.500000  20.050000   
199  34.422727  86.727273     7.500000    80.000000  16.186364   
343  33.864909  88.456364     4.457059    81.893529  14.962545   
462  34.055556  84.555556     4.600000    85.500000  15.877778   
580  32.200000  93.266667     2.500000    89.775000  16.666667   

     Red Blood Cells  White Blood Cells  Monocytes.2  Lymphocytes.2  \
65          3.042500           9.937500          NaN            NaN   
199         3.645455          39.627273          NaN            NaN   
343         3.485691          10.209091          NaN            NaN   
462         3.636667           9.055556          NaN            NaN   
580         2.871333           9.873333          NaN            NaN   

     Monocytes.3  Eosinophils.2  Lymphocytes.3  Eosinophils.3  Lymphocytes.4  \
65           NaN            NaN            NaN            NaN            NaN   
199          NaN            NaN            NaN            NaN            NaN   
343          NaN            NaN            NaN            NaN            NaN   
462          NaN            NaN            NaN            NaN            NaN   
580          NaN            NaN            NaN            1.0           90.0   

    gender                  dob                  dod             dod_hosp  \
65       F  1895-05-17 00:00:00  2195-11-24 00:00:00                  NaN   
199      F  2061-03-25 00:00:00  2129-12-01 00:00:00  2129-12-01 00:00:00   
343      M  2029-07-09 00:00:00  2107-07-20 00:00:00                  NaN   
462      F  2099-03-17 00:00:00  2178-05-02 00:00:00  2178-05-02 00:00:00   
580      F  2063-07-05 00:00:00  2130-11-03 00:00:00  2130-11-03 00:00:00   

                 dod_ssn  expire_flag short_title           long_title  
65   2195-11-24 00:00:00            1  Anemia NOS  Anemia, unspecified  
199  2129-12-01 00:00:00            1  Anemia NOS  Anemia, unspecified  
343  2107-07-20 00:00:00            1  Anemia NOS  Anemia, unspecified  
462  2178-05-02 00:00:00            1  Anemia NOS  Anemia, unspecified  
580  2130-11-03 00:00:00            1  Anemia NOS  Anemia, unspecified

In [50]:
selected_columns_df = Anemia_cbc_dataset_diagnoses[['Neutrophils','Monocytes.1','Hematocrit']]
Anemia_cbc_dataset_diagnoses = selected_columns_df

In [51]:
Anemia_cbc_dataset_diagnoses.describe()

Neutrophils  Monocytes.1  Hematocrit
count    14.000000    14.000000   17.000000
mean     81.112709     4.966852   29.001125
std       8.474870     1.647380    3.471754
min      67.100000     2.000000   24.233333
25%      75.100000     4.467794   26.269231
50%      81.867146     4.685294   27.975000
75%      86.613971     6.000000   31.098467
max      95.900000     7.600000   36.500000

------

In [52]:
cbc_diseased_df
cbc_non_diseased_df

index  S.No.   Age  Sex   RBC   PCV    MCV   MCH  MCHC   RDW    TLC  \
1      1.0    2.0  41.0  0.0  4.78  44.5   93.1  28.9  31.0  13.0   7.02   
2      2.0    3.0  40.0  1.0  4.65  41.6   89.5  28.8  32.2  13.0   8.09   
4      4.0    5.0  20.0  1.0  4.14  36.9   89.1  27.8  31.2  13.2   4.75   
9     11.0   12.0  28.0  0.0  4.81  44.4   92.3  27.9  30.2  14.3   6.22   
13    15.0   16.0  22.0  0.0  5.17  44.5   86.1  27.7  32.1  13.2  10.70   
14    16.0   17.0  64.0  0.0  4.60  41.4   90.0  28.5  31.6  14.4   9.67   
38    41.0   42.0  63.0  0.0  4.67  43.6   93.4  31.7  33.9  13.5   9.80   
40    43.0   44.0  68.0  0.0  4.57  39.1   85.6  27.6  32.2  13.9   6.60   
47    50.0   51.0  42.0  0.0  4.99  44.6   89.4  28.3  31.6  13.9   6.11   
48    51.0   52.0  58.0  1.0  4.18  36.3   84.4  26.3  31.2  13.6   9.74   
49    52.0   53.0  60.0  0.0  4.74  41.8   87.8  28.3  32.2  13.3   9.06   
50    53.0   54.0  40.0  1.0  4.59  38.3   83.3  26.1  31.3  14.9   7.55   
55    62.0   63.0  35.0  1.0  4.30  41.5   95.5  30.7  31.8  13.6   6.47   
59    67.0   68.0  32.0  0.0  5.01  46.5   92.8  29.1  31.4  14.0   5.40   
62    73.0   74.0  44.0  1.0  4.90  38.3   78.2  25.9  33.2  13.6   9.10   
64    75.0   76.0  24.0  1.0  4.47  36.0   80.5  25.3  31.1  14.8   8.00   
76    88.0   89.0  43.0  0.0  5.18  44.9   86.7  29.0  33.4  13.3   9.30   
80    92.0   93.0  35.0  0.0  4.57  41.6   91.0  30.9  33.9  14.1   5.50   
84    96.0   97.0  35.0  1.0  4.10  38.9   94.9  28.8  30.3  12.7   5.84   
85    97.0   98.0  24.0  0.0  4.21  42.6  101.2  32.3  31.9  15.0   7.61   
92   104.0  105.0  32.0  0.0  4.76  40.3   84.7  28.2  33.3  12.9   4.20   
109  122.0  123.0  49.0  0.0  4.79  43.0   89.8  26.8  32.1  13.1   5.44   
110  123.0  124.0  42.0  1.0  4.22  35.5   84.1  27.3  32.4  13.1   6.50   
113  126.0  127.0  52.0  0.0  4.54  43.5   95.8  29.7  31.0  14.7   8.68   
116  130.0  131.0  35.0  1.0  4.10  38.6   94.9  26.8  30.3  12.7   5.84   
120  134.0  135.0  35.0  0.0  4.64  40.5   87.3  28.7  32.8  12.8   8.98   
124  139.0  140.0  52.0  0.0  4.54  43.5   95.8  29.7  31.0  14.7   8.68   
125  140.0  141.0  42.0  1.0  4.22  35.5   84.1  27.3  32.4  13.1   6.50   
132  147.0  148.0  35.0  1.0  4.10  38.9   94.9  28.8  30.3  12.7   5.84   
133  148.0  149.0  30.0  0.0  4.95  43.6   88.1  26.9  30.5  14.0   7.38   
175  194.0  195.0  31.0  0.0  4.46  44.9  100.7  33.6  33.4  14.7   8.45   
176  195.0  196.0  72.0  0.0  4.91  44.1   89.8  28.3  31.5  14.1   8.76   
179  199.0  200.0  53.0  0.0  4.16  38.8   93.3  29.1  31.2  13.3  10.30   
180  200.0  201.0  38.0  1.0  4.61  39.1   84.8  26.7  31.5  12.9   9.34   
184  204.0  205.0  65.0  0.0  4.55  40.4   88.8  28.8  32.4  13.9   4.50   
185  205.0  206.0  20.0  0.0  4.62  43.2   93.5  29.7  31.7  15.9   4.29   
188  208.0  209.0  29.0  0.0  4.78  42.8   89.5  27.4  30.6  14.3   6.40   
190  210.0  211.0  66.0  0.0  4.57  39.9   87.3  27.1  31.1  14.2   7.05   
194  215.0  216.0  38.0  1.0  4.61  39.1   84.8  26.7  31.5  12.9   9.34   
213  235.0  236.0  65.0  0.0  4.70  41.3   87.9  27.4  31.2  14.1   9.87   
215  237.0  238.0  74.0  0.0  4.13  40.2   97.3  30.5  31.3  15.3   8.54   
218  240.0  241.0  65.0  0.0  4.76  41.8   87.8  28.2  32.1  12.6   7.56   
242  264.0  265.0  33.0  0.0  5.03  42.4   84.3  25.8  30.4  14.6   6.01   
246  268.0  269.0  30.0  1.0  4.48  42.1   94.0  28.8  30.6  14.6   7.12   
247  269.0  270.0  36.0  0.0  4.71  40.3   85.6  25.7  30.0  14.9   5.02   
252  274.0  275.0  63.0  0.0  5.53  46.2   83.5  25.1  30.1  15.3   7.97   
254  276.0  277.0  25.0  1.0  4.86  42.0   86.4  26.5  30.7  13.7   8.13   
256  278.0  279.0  24.0  0.0  5.17  47.4   91.7  28.2  30.8  13.2   8.27   
263  286.0  287.0  30.0  0.0  4.80  43.4   90.4  28.8  31.8  12.7   6.10   
287  311.0  312.0  19.0  0.0  5.08  45.9   90.4  28.1  31.2  14.7   4.64   
305  330.0  331.0  47.0  0.0  4.61  40.9   88.8  29.2  32.9  14.0   6.30   
308  333.0  334.0  26.0  1.0  4.36  38.1   87.4  29.6

In [53]:
cbc_diseased_df.head()

index  S.No.   Age  Sex   RBC   PCV   MCV   MCH  MCHC   RDW    TLC  \
0    0.0    1.0  28.0  0.0  5.66  34.0  60.1  17.0  28.2  20.0  11.10   
3    3.0    4.0  76.0  0.0  4.24  36.7  86.6  26.7  30.8  14.9  13.41   
5    5.0    6.0  24.0  0.0  4.29  40.1  93.5  29.6  31.7  14.5  13.96   
6    6.0    7.0  28.0  1.0  4.98  42.3  84.9  24.9  29.3  16.2   9.33   
7    9.0   10.0  62.0  0.0  5.25  45.6  86.9  25.3  29.2  15.6  10.68   

   PLT/mm3   HGB  diseased  
0    128.3   9.6       1.0  
3    264.0  11.3       1.0  
5    233.0  12.7       1.0  
6    213.0  12.4       1.0  
7    151.0  13.3       1.0

In [54]:
cbc_diseased_df.columns

Index(['index', 'S.No.', 'Age', 'Sex', 'RBC', 'PCV', 'MCV', 'MCH', 'MCHC',
       'RDW', 'TLC', 'PLT/mm3', 'HGB', 'diseased'],
      dtype='object')

In [55]:
selected_columns_df_dis = cbc_diseased_df[['Age','Sex','RBC','HGB','TLC','PLT/mm3']]
selected_columns_df_dis_non = cbc_non_diseased_df[['Age','Sex','RBC','HGB','TLC','PLT/mm3']]
new_cbc_non_diseased_df = selected_columns_df_dis_non
new_cbc_diseased_df = selected_columns_df_dis



In [56]:
new_cbc_diseased_df['Neutrophil'] = Anemia_cbc_dataset_diagnoses['Neutrophils'].mean()
new_cbc_diseased_df['Monocyte'] = Anemia_cbc_dataset_diagnoses['Monocytes.1'].mean()
new_cbc_diseased_df['Hematocrit'] = Anemia_cbc_dataset_diagnoses['Hematocrit'].mean()
new_cbc_non_diseased_df['Hematocrit'] = new_cbc_dataset_kenya['Hct'].mean()
new_cbc_non_diseased_df['Neutrophil'] = new_cbc_dataset_kenya['Neutrophil %'].mean()
new_cbc_non_diseased_df['Monocyte'] = new_cbc_dataset_kenya['Monocyte %'].mean()

C:\Users\Bravo\AppData\Local\Temp\ipykernel_17800\632986505.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  new_cbc_diseased_df['Neutrophil'] = Anemia_cbc_dataset_diagnoses['Neutrophils'].mean()
C:\Users\Bravo\AppData\Local\Temp\ipykernel_17800\632986505.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  new_cbc_diseased_df['Monocyte'] = Anemia_cbc_dataset_diagnoses['Monocytes.1'].mean()
C:\Users\Bravo\AppData\Local\Temp\ipykernel_17800\632986505.py:3: SettingWithCopyWarning: 
A value is trying to be s

In [57]:
new_cbc_diseased_df.rename(columns={'TLC': 'WBC'}, inplace=True)
new_cbc_non_diseased_df.rename(columns={'TLC': 'WBC'}, inplace=True)



C:\Users\Bravo\AppData\Local\Temp\ipykernel_17800\1978585002.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  new_cbc_diseased_df.rename(columns={'TLC': 'WBC'}, inplace=True)
C:\Users\Bravo\AppData\Local\Temp\ipykernel_17800\1978585002.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  new_cbc_non_diseased_df.rename(columns={'TLC': 'WBC'}, inplace=True)


In [58]:
new_cbc_diseased_df.drop(columns=['RBC'],axis=1, inplace=True)
new_cbc_non_diseased_df.drop(columns=['RBC'], axis=1, inplace=True)


C:\Users\Bravo\AppData\Local\Temp\ipykernel_17800\250116158.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  new_cbc_diseased_df.drop(columns=['RBC'],axis=1, inplace=True)
C:\Users\Bravo\AppData\Local\Temp\ipykernel_17800\250116158.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  new_cbc_non_diseased_df.drop(columns=['RBC'], axis=1, inplace=True)


In [59]:
new_cbc_non_diseased_df['Diagnosis'] = 'normal'
new_cbc_diseased_df['Diagnosis'] = 'anemia'

C:\Users\Bravo\AppData\Local\Temp\ipykernel_17800\146706923.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  new_cbc_non_diseased_df['Diagnosis'] = 'normal'
C:\Users\Bravo\AppData\Local\Temp\ipykernel_17800\146706923.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  new_cbc_diseased_df['Diagnosis'] = 'anemia'


In [60]:
new_cbc_non_diseased_df.head()


Age  Sex   HGB    WBC  PLT/mm3  Hematocrit  Neutrophil  Monocyte  \
1   41.0  0.0  13.8   7.02    419.0    0.452955   45.342235  6.828409   
2   40.0  1.0  13.4   8.09    325.0    0.452955   45.342235  6.828409   
4   20.0  1.0  11.5   4.75    196.0    0.452955   45.342235  6.828409   
9   28.0  0.0  13.4   6.22    150.0    0.452955   45.342235  6.828409   
13  22.0  0.0  14.3  10.70    268.0    0.452955   45.342235  6.828409   

   Diagnosis  
1     normal  
2     normal  
4     normal  
9     normal  
13    normal

In [61]:
new_cbc_diseased_df.head()

Age  Sex   HGB    WBC  PLT/mm3  Neutrophil  Monocyte  Hematocrit Diagnosis
0  28.0  0.0   9.6  11.10    128.3   81.112709  4.966852   29.001125    anemia
3  76.0  0.0  11.3  13.41    264.0   81.112709  4.966852   29.001125    anemia
5  24.0  0.0  12.7  13.96    233.0   81.112709  4.966852   29.001125    anemia
6  28.0  1.0  12.4   9.33    213.0   81.112709  4.966852   29.001125    anemia
7  62.0  0.0  13.3  10.68    151.0   81.112709  4.966852   29.001125    anemia

 ---------------


 ---------------


 ---------------


In [62]:
print(new_cbc_non_diseased_df.columns)
print(new_cbc_diseased_df.columns)
print(new_cbc_dataset_kenya.columns)
print(new_cbc_dataset_dengue.columns)
print(new_cbc_sav_non_nvp_df.columns)




Index(['Age', 'Sex', 'HGB', 'WBC', 'PLT/mm3', 'Hematocrit', 'Neutrophil',
       'Monocyte', 'Diagnosis'],
      dtype='object')
Index(['Age', 'Sex', 'HGB', 'WBC', 'PLT/mm3', 'Neutrophil', 'Monocyte',
       'Hematocrit', 'Diagnosis'],
      dtype='object')
Index(['Age', 'Sex', 'Hb', 'Hct', 'WBC', 'Platelets', 'Neutrophil %',
       'Monocyte %', 'Diagnosis'],
      dtype='object')
Index(['age', 'sex', 'wbc', 'neut%', 'mono%', 'hb', 'plt', 'hct', 'Diagnosis'], dtype='object')
Index(['YAŞ', 'HGB', 'HCT', 'WBC', 'PLT', 'NÖTROFİL', 'MONOSİT', 'sex',
       'Diagnosis'],
      dtype='object')


In [63]:
# สลับลำดับของคอลัมน์ใน DataFrame
new_cbc_diseased_df = new_cbc_diseased_df[['Age', 'Sex', 'HGB', 'WBC', 'PLT/mm3', 'Hematocrit', 'Neutrophil','Monocyte', 'Diagnosis']]


In [64]:
new_cbc_non_diseased_df = new_cbc_non_diseased_df[['Age', 'Sex', 'HGB', 'WBC', 'PLT/mm3', 'Hematocrit', 'Neutrophil','Monocyte', 'Diagnosis']]

In [65]:
new_cbc_dataset_kenya = new_cbc_dataset_kenya[['Age', 'Sex', 'Hb', 'WBC', 'Platelets','Hct', 'Neutrophil %','Monocyte %', 'Diagnosis']]

In [66]:
new_cbc_dataset_dengue = new_cbc_dataset_dengue[['age', 'sex', 'hb', 'wbc', 'plt','hct','neut%', 'mono%',   'Diagnosis']]

In [67]:
new_cbc_sav_non_nvp_df = new_cbc_sav_non_nvp_df[['YAŞ', 'sex','HGB','WBC', 'PLT','HCT',   'NÖTROFİL', 'MONOSİT', 'Diagnosis']]

In [68]:
print(new_cbc_non_diseased_df.columns)
print(new_cbc_diseased_df.columns)
print(new_cbc_dataset_kenya.columns)
print(new_cbc_dataset_dengue.columns)
print(new_cbc_sav_non_nvp_df.columns)

Index(['Age', 'Sex', 'HGB', 'WBC', 'PLT/mm3', 'Hematocrit', 'Neutrophil',
       'Monocyte', 'Diagnosis'],
      dtype='object')
Index(['Age', 'Sex', 'HGB', 'WBC', 'PLT/mm3', 'Hematocrit', 'Neutrophil',
       'Monocyte', 'Diagnosis'],
      dtype='object')
Index(['Age', 'Sex', 'Hb', 'WBC', 'Platelets', 'Hct', 'Neutrophil %',
       'Monocyte %', 'Diagnosis'],
      dtype='object')
Index(['age', 'sex', 'hb', 'wbc', 'plt', 'hct', 'neut%', 'mono%', 'Diagnosis'], dtype='object')
Index(['YAŞ', 'sex', 'HGB', 'WBC', 'PLT', 'HCT', 'NÖTROFİL', 'MONOSİT',
       'Diagnosis'],
      dtype='object')


In [69]:
new_cbc_non_diseased_df.rename(columns={'PLT/mm3': 'PLT','Hematocrit':'HCT','Neutrophil':'Neutro','Monocyte':'Mono'}, inplace=True)
new_cbc_diseased_df.rename(columns={'PLT/mm3': 'PLT','Hematocrit':'HCT','Neutrophil':'Neutro','Monocyte':'Mono'}, inplace=True)
new_cbc_dataset_kenya.rename(columns={'Hb':'HGB','Neutrophil %': 'Neutro','Hct':'HCT','Platelets': 'PLT','Monocyte %':'Mono'}, inplace=True)
new_cbc_dataset_dengue.rename(columns={'age':'Age','sex': 'Sex','hb':'HGB','wbc':'WBC','plt':'PLT','hct':'HCT','neut%':'Neutro','mono%':'Mono'}, inplace=True)
new_cbc_sav_non_nvp_df.rename(columns={'YAŞ':'Age','sex': 'Sex','NÖTROFİL':'Neutro','MONOSİT':'Mono'}, inplace=True)


In [70]:
print(new_cbc_non_diseased_df.columns)
print(new_cbc_diseased_df.columns)
print(new_cbc_dataset_kenya.columns)
print(new_cbc_dataset_dengue.columns)
print(new_cbc_sav_non_nvp_df.columns)

Index(['Age', 'Sex', 'HGB', 'WBC', 'PLT', 'HCT', 'Neutro', 'Mono',
       'Diagnosis'],
      dtype='object')
Index(['Age', 'Sex', 'HGB', 'WBC', 'PLT', 'HCT', 'Neutro', 'Mono',
       'Diagnosis'],
      dtype='object')
Index(['Age', 'Sex', 'HGB', 'WBC', 'PLT', 'HCT', 'Neutro', 'Mono',
       'Diagnosis'],
      dtype='object')
Index(['Age', 'Sex', 'HGB', 'WBC', 'PLT', 'HCT', 'Neutro', 'Mono',
       'Diagnosis'],
      dtype='object')
Index(['Age', 'Sex', 'HGB', 'WBC', 'PLT', 'HCT', 'Neutro', 'Mono',
       'Diagnosis'],
      dtype='object')


In [71]:
print(new_cbc_non_diseased_df.head())
print(new_cbc_diseased_df.head())
print(new_cbc_dataset_kenya.head())
print(new_cbc_dataset_dengue.head())
print(new_cbc_sav_non_nvp_df.head())

     Age  Sex   HGB    WBC    PLT       HCT     Neutro      Mono Diagnosis
1   41.0  0.0  13.8   7.02  419.0  0.452955  45.342235  6.828409    normal
2   40.0  1.0  13.4   8.09  325.0  0.452955  45.342235  6.828409    normal
4   20.0  1.0  11.5   4.75  196.0  0.452955  45.342235  6.828409    normal
9   28.0  0.0  13.4   6.22  150.0  0.452955  45.342235  6.828409    normal
13  22.0  0.0  14.3  10.70  268.0  0.452955  45.342235  6.828409    normal
    Age  Sex   HGB    WBC    PLT        HCT     Neutro      Mono Diagnosis
0  28.0  0.0   9.6  11.10  128.3  29.001125  81.112709  4.966852    anemia
3  76.0  0.0  11.3  13.41  264.0  29.001125  81.112709  4.966852    anemia
5  24.0  0.0  12.7  13.96  233.0  29.001125  81.112709  4.966852    anemia
6  28.0  1.0  12.4   9.33  213.0  29.001125  81.112709  4.966852    anemia
7  62.0  0.0  13.3  10.68  151.0  29.001125  81.112709  4.966852    anemia
   Age  Sex   HGB   WBC  PLT   HCT  Neutro  Mono Diagnosis
0   23    1  15.9  6.16  198  0.47    48.

In [72]:
print(new_cbc_non_diseased_df.info())
print(new_cbc_diseased_df.info())
print(new_cbc_dataset_kenya.info())
print(new_cbc_dataset_dengue.info())
print(new_cbc_sav_non_nvp_df.info())

<class 'pandas.core.frame.DataFrame'>
Index: 63 entries, 1 to 336
Data columns (total 9 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Age        63 non-null     float64
 1   Sex        63 non-null     float64
 2   HGB        63 non-null     float64
 3   WBC        63 non-null     float64
 4   PLT        63 non-null     float64
 5   HCT        63 non-null     float64
 6   Neutro     63 non-null     float64
 7   Mono       63 non-null     float64
 8   Diagnosis  63 non-null     object 
dtypes: float64(8), object(1)
memory usage: 4.9+ KB
None
<class 'pandas.core.frame.DataFrame'>
Index: 275 entries, 0 to 337
Data columns (total 9 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Age        275 non-null    float64
 1   Sex        275 non-null    float64
 2   HGB        275 non-null    float64
 3   WBC        275 non-null    float64
 4   PLT        275 non-null    float64
 5   HCT        275 non-nu

In [73]:
# เปลี่ยนค่าในคอลัมน์ 'Sex' ของ DataFrame เพื่อแปลงค่าที่เท่ากับ 1 เป็น 'male' และค่าเท่ากับ 0 เป็น 'female'
new_cbc_non_diseased_df['Sex'] = new_cbc_non_diseased_df['Sex'].replace({1: 'male', 0: 'female'})
new_cbc_diseased_df['Sex'] = new_cbc_diseased_df['Sex'].replace({1: 'male', 0: 'female'})
new_cbc_dataset_kenya['Sex'] = new_cbc_dataset_kenya['Sex'].replace({1: 'male', 0: 'female'})


In [74]:
print(new_cbc_non_diseased_df.info())
print(new_cbc_diseased_df.info())
print(new_cbc_dataset_kenya.info())
print(new_cbc_dataset_dengue.info())
print(new_cbc_sav_non_nvp_df.info())

<class 'pandas.core.frame.DataFrame'>
Index: 63 entries, 1 to 336
Data columns (total 9 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Age        63 non-null     float64
 1   Sex        63 non-null     object 
 2   HGB        63 non-null     float64
 3   WBC        63 non-null     float64
 4   PLT        63 non-null     float64
 5   HCT        63 non-null     float64
 6   Neutro     63 non-null     float64
 7   Mono       63 non-null     float64
 8   Diagnosis  63 non-null     object 
dtypes: float64(7), object(2)
memory usage: 4.9+ KB
None
<class 'pandas.core.frame.DataFrame'>
Index: 275 entries, 0 to 337
Data columns (total 9 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Age        275 non-null    float64
 1   Sex        275 non-null    object 
 2   HGB        275 non-null    float64
 3   WBC        275 non-null    float64
 4   PLT        275 non-null    float64
 5   HCT        275 non-nu

In [75]:
# เปลี่ยนประเภทของคอลัมน์เป็น float
new_cbc_dataset_kenya[['Age', 'PLT']] = new_cbc_dataset_kenya[['Age', 'PLT']].astype(float)
new_cbc_dataset_dengue[['Age', 'HGB', 'PLT']] = new_cbc_dataset_dengue[['Age', 'HGB', 'PLT']].astype(float)


In [76]:
print(new_cbc_non_diseased_df.info())
print(new_cbc_diseased_df.info())
print(new_cbc_dataset_kenya.info())
print(new_cbc_dataset_dengue.info())
print(new_cbc_sav_non_nvp_df.info())

<class 'pandas.core.frame.DataFrame'>
Index: 63 entries, 1 to 336
Data columns (total 9 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Age        63 non-null     float64
 1   Sex        63 non-null     object 
 2   HGB        63 non-null     float64
 3   WBC        63 non-null     float64
 4   PLT        63 non-null     float64
 5   HCT        63 non-null     float64
 6   Neutro     63 non-null     float64
 7   Mono       63 non-null     float64
 8   Diagnosis  63 non-null     object 
dtypes: float64(7), object(2)
memory usage: 4.9+ KB
None
<class 'pandas.core.frame.DataFrame'>
Index: 275 entries, 0 to 337
Data columns (total 9 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   Age        275 non-null    float64
 1   Sex        275 non-null    object 
 2   HGB        275 non-null    float64
 3   WBC        275 non-null    float64
 4   PLT        275 non-null    float64
 5   HCT        275 non-nu

In [77]:
print(new_cbc_non_diseased_df.head())
print(new_cbc_diseased_df.head())
print(new_cbc_dataset_kenya.head())
print(new_cbc_dataset_dengue.head())
print(new_cbc_sav_non_nvp_df.head())

     Age     Sex   HGB    WBC    PLT       HCT     Neutro      Mono Diagnosis
1   41.0  female  13.8   7.02  419.0  0.452955  45.342235  6.828409    normal
2   40.0    male  13.4   8.09  325.0  0.452955  45.342235  6.828409    normal
4   20.0    male  11.5   4.75  196.0  0.452955  45.342235  6.828409    normal
9   28.0  female  13.4   6.22  150.0  0.452955  45.342235  6.828409    normal
13  22.0  female  14.3  10.70  268.0  0.452955  45.342235  6.828409    normal
    Age     Sex   HGB    WBC    PLT        HCT     Neutro      Mono Diagnosis
0  28.0  female   9.6  11.10  128.3  29.001125  81.112709  4.966852    anemia
3  76.0  female  11.3  13.41  264.0  29.001125  81.112709  4.966852    anemia
5  24.0  female  12.7  13.96  233.0  29.001125  81.112709  4.966852    anemia
6  28.0    male  12.4   9.33  213.0  29.001125  81.112709  4.966852    anemia
7  62.0  female  13.3  10.68  151.0  29.001125  81.112709  4.966852    anemia
    Age     Sex   HGB   WBC    PLT   HCT  Neutro  Mono Diagnosis

------------------

In [78]:
new_cbc_non_diseased_df['HCT'] *= 100

new_cbc_dataset_kenya['HCT'] *= 100

new_cbc_dataset_dengue['HGB'] /= 10

new_cbc_dataset_dengue['PLT'] *= 10

new_cbc_sav_non_nvp_df['WBC'] /= 1000

new_cbc_sav_non_nvp_df['PLT'] /= 1000

new_cbc_sav_non_nvp_df['Neutro'] /= 100

new_cbc_sav_non_nvp_df['Mono'] /= 100


In [79]:

new_cbc_non_diseased_df['Neutro'] /= 10

new_cbc_diseased_df['Neutro'] /= 10

new_cbc_dataset_kenya['Neutro'] /= 10

new_cbc_dataset_dengue['Neutro'] /= 10

new_cbc_sav_non_nvp_df['Neutro'] /= 10

In [80]:
print(new_cbc_non_diseased_df.head())
print(new_cbc_diseased_df.head())
print(new_cbc_dataset_kenya.head())
print(new_cbc_dataset_dengue.head())
print(new_cbc_sav_non_nvp_df.head())

     Age     Sex   HGB    WBC    PLT        HCT    Neutro      Mono Diagnosis
1   41.0  female  13.8   7.02  419.0  45.295455  4.534223  6.828409    normal
2   40.0    male  13.4   8.09  325.0  45.295455  4.534223  6.828409    normal
4   20.0    male  11.5   4.75  196.0  45.295455  4.534223  6.828409    normal
9   28.0  female  13.4   6.22  150.0  45.295455  4.534223  6.828409    normal
13  22.0  female  14.3  10.70  268.0  45.295455  4.534223  6.828409    normal
    Age     Sex   HGB    WBC    PLT        HCT    Neutro      Mono Diagnosis
0  28.0  female   9.6  11.10  128.3  29.001125  8.111271  4.966852    anemia
3  76.0  female  11.3  13.41  264.0  29.001125  8.111271  4.966852    anemia
5  24.0  female  12.7  13.96  233.0  29.001125  8.111271  4.966852    anemia
6  28.0    male  12.4   9.33  213.0  29.001125  8.111271  4.966852    anemia
7  62.0  female  13.3  10.68  151.0  29.001125  8.111271  4.966852    anemia
    Age     Sex   HGB   WBC    PLT   HCT  Neutro  Mono Diagnosis
0  23

--------

In [81]:
Data_df = pd.concat([new_cbc_non_diseased_df, new_cbc_diseased_df,new_cbc_dataset_kenya,new_cbc_dataset_dengue,new_cbc_sav_non_nvp_df], axis=0)


In [82]:
Data_df = Data_df.reset_index(drop=True)


In [83]:
pd.set_option('display.max_rows', 100)
print(Data_df)

       Age     Sex   HGB    WBC    PLT        HCT    Neutro      Mono  \
0     41.0  female  13.8   7.02  419.0  45.295455  4.534223  6.828409   
1     40.0    male  13.4   8.09  325.0  45.295455  4.534223  6.828409   
2     20.0    male  11.5   4.75  196.0  45.295455  4.534223  6.828409   
3     28.0  female  13.4   6.22  150.0  45.295455  4.534223  6.828409   
4     22.0  female  14.3  10.70  268.0  45.295455  4.534223  6.828409   
...    ...     ...   ...    ...    ...        ...       ...       ...   
2619  33.0  female  13.8   7.87  299.0  40.900000  6.000000  3.700000   
2620  23.0  female  14.2  11.20  226.0  41.400000  9.370000  4.700000   
2621  26.0  female  13.1   9.25  267.0  38.600000  7.070000  3.800000   
2622  42.0  female  13.6   8.26  199.0  40.000000  6.700000  3.600000   
2623  22.0  female  13.6  11.62  353.0  38.000000  9.560000  5.600000   

     Diagnosis  
0       normal  
1       normal  
2       normal  
3       normal  
4       normal  
...        ...  
2619

In [96]:
Data_df['Neutro'] *= 10
Data_df['PLT'] *= 1000
Data_df['WBC'] *= 1000


In [97]:
Data_df.head()

Age     Sex   HGB      WBC       PLT        HCT     Neutro      Mono  \
0  41.0  female  13.8   7020.0  419000.0  45.295455  45.342235  6.828409   
1  40.0    male  13.4   8090.0  325000.0  45.295455  45.342235  6.828409   
2  20.0    male  11.5   4750.0  196000.0  45.295455  45.342235  6.828409   
3  28.0  female  13.4   6220.0  150000.0  45.295455  45.342235  6.828409   
4  22.0  female  14.3  10700.0  268000.0  45.295455  45.342235  6.828409   

  Diagnosis  
0    normal  
1    normal  
2    normal  
3    normal  
4    normal

In [98]:
sum_of_nans = Data_df.isna().sum()
print(sum_of_nans)

Age          0
Sex          0
HGB          0
WBC          0
PLT          0
HCT          0
Neutro       0
Mono         0
Diagnosis    0
dtype: int64


 ---------------


----------------


   <p>
Packed Cell Volume (PCV) *drop<p> 
 Neutrophils *drop<p>
 Lymphocytes *drop<p>
 มีผลเฉย<p>
 <p>
โรคไข้เลือดออก<p>
WBC: ปกติหรือต่ำ (มักจะต่ำ)<p>
Neutrophils: ปกติหรือต่ำ<p>
Lymphocytes: ปกติหรือสูง<p>
<p>
โรคโลหิตจาง<p>
WBC: ปกติหรือต่ำ (ขึ้นอยู่กับสาเหตุ)<p>
Neutrophils: ปกติหรือต่ำ (ขึ้นอยู่กับสาเหตุ)<p>
Lymphocytes: ปกติหรือต่ำ (ขึ้นอยู่กับสาเหตุ)<p>

 ---------------



นำ meandeley ไป เติมค่า WBC โดยอิงจาก ไม่เป็นเลือดจาง - kenya (แยก เพศ เติมด้วยค่าเฉลี่ย) <p>
นำ meandeley ไป เติมค่า WBC โดยอิงจาก เป็นเลือดจาง - kenya (แยก เพศ เติมด้วยmin...) 


 ---------------


input   X <p>
ผลการตรวจเลือด (CBC)  <p>
output  Y <p>
ทำนายว่า เลือด ปกติ หรือ เป็นไข้เลือดออกที่ระดับ ปกติ-อันตรายไหม


 ---------------


มี 2 กลุ่ม <p>
1.กลุ่มผู้ติดเชื้อ  
 `cbc_dataset_dengue` 1984 rows
<li>กลุ่มผู้ติดเชื้อระดับปกติ
<li>กลุ่มผู้ติดเชื้อระดับอันตราย
<li>กลุ่มผู้ติดเชื้ออื่นๆ
<li>กลุ่มผู้ต้องสงสัยว่าติดเชื้ออื่นๆ

<p>

2.กลุ่มผู้มีโรคโลหิตจาง <p>
`cbc_dataset_meandeley` 338 rows<p>
<li>โดยมีคนโรคโลหิตจาง 275 rows


3.กลุ่มผู้มีสุขภาพปกติ <p>
`cbc_dataset_kenya` 528 rows <p>
`cbc_dataset_sav` 278 rows <p>
`cbc_dataset_meandeley` 338 rows<p>
<li>โดยมีคนสุขภาพปกติ 63 rows

*option* <p>
4.กลุ่มผู้ที่ไม่มีข้อมูลสุขภาพ(เอาไว้ทำนาย ในสถานการณ์จริง)  <p> 
`cbc_dataset_infomation` 500 rows<p> 
`cbc_dataset_pregnant` 869 rows

 ---------------


In [ ]:
Data_df.columns

Index(['Age', 'Sex', 'HGB', 'WBC', 'PLT', 'HCT', 'Neutro', 'Mono',
       'Diagnosis'],
      dtype='object')

In [100]:
from sklearn.preprocessing import StandardScaler

scale_columns = ['HGB', 'WBC','PLT','HCT','Neutro','Mono']

# คัดลอก DataFrame เพื่อทำ StandardScaler
Data_df_stdScaler = Data_df.copy()

# สร้าง StandardScaler 
scaler = StandardScaler()
Data_df_stdScaler[scale_columns] = scaler.fit_transform(Data_df[scale_columns])


In [101]:
Data_df_stdScaler.head()

Age     Sex       HGB       WBC       PLT       HCT    Neutro      Mono  \
0  41.0  female  0.251887 -0.109213 -0.568371  0.666063 -0.613704 -0.237428   
1  40.0    male  0.077459 -0.021415 -0.637539  0.666063 -0.613704 -0.237428   
2  20.0    male -0.751075 -0.295476 -0.732459  0.666063 -0.613704 -0.237428   
3  28.0  female  0.077459 -0.174856 -0.766307  0.666063 -0.613704 -0.237428   
4  22.0  female  0.469923  0.192747 -0.679480  0.666063 -0.613704 -0.237428   

  Diagnosis  
0    normal  
1    normal  
2    normal  
3    normal  
4    normal

-----

In [104]:
Data_df_stdScaler.to_csv('Data_df_stdScaler.csv', index=False)

In [105]:
check_Data = pd.read_csv("./Data_df_stdScaler.csv")
check_Data.head()

Age     Sex       HGB       WBC       PLT       HCT    Neutro      Mono  \
0  41.0  female  0.251887 -0.109213 -0.568371  0.666063 -0.613704 -0.237428   
1  40.0    male  0.077459 -0.021415 -0.637539  0.666063 -0.613704 -0.237428   
2  20.0    male -0.751075 -0.295476 -0.732459  0.666063 -0.613704 -0.237428   
3  28.0  female  0.077459 -0.174856 -0.766307  0.666063 -0.613704 -0.237428   
4  22.0  female  0.469923  0.192747 -0.679480  0.666063 -0.613704 -0.237428   

  Diagnosis  
0    normal  
1    normal  
2    normal  
3    normal  
4    normal